mangaki/mangaki/admin.py

```diff
@@ -139,29 +142,97 @@ def make_sfw(self, request, queryset):
 
     make_sfw.short_description = "Rendre SFW les œuvres sélectionnées"
 
+    @transaction.atomic  # In case trouble happens
     def merge(self, request, queryset):
         queryset = queryset.order_by('id')
-        if request.POST.get('post'):
-            chosen_id = int(request.POST.get('chosen_id'))
-            for obj in queryset:
-                if obj.id != chosen_id:
-                    for rating in Rating.objects.filter(work=obj).select_related('user'):
-                        # S'il n'a pas déjà voté pour l'autre
+        if request.POST.get('confirm'):
+            chosen_id = int(request.POST.get('id'))
+            chosen_work = Work.objects.get(id=chosen_id)
+            for field in request.POST.get('fields_to_choose').split(','):
+                if request.POST.get(field) != 'None':
+                    setattr(chosen_work, field, request.POST.get(field))
+            for work in queryset:
+                if work.id != chosen_id:
+                    for rating in work.rating_set.select_related('user'):
                         if Rating.objects.filter(user=rating.user, work__id=chosen_id).count() == 0:
+                            # No rating already exists for this work
                             rating.work_id = chosen_id
                             rating.save()
                         else:
+                            other_rating = Rating.objects.get(user=rating.user, work__id=chosen_id)
```

Comment:
If they're unique, just use get and catch the exception when it does not exists.

It's unnecessary to check for existence when get will do it.

But you must be sure it's unique. Otherwise, MultipleObjectsReturned can be thrown also.